samples/java-eventing-shopping-cart/src/main/java/shopping/cart/ShoppingCartView2.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package shopping.cart;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import shopping.cart.domain.ShoppingCartDomain;
+import shopping.cart.view.ShoppingCartViewModel;
+
+import java.util.Optional;
+
+// user view impl
+public class ShoppingCartView2 extends AbstractShoppingCartView {
+
+  private static Logger LOG = LoggerFactory.getLogger(ShoppingCartView2.class);
+
+  public UpdateEffect<ShoppingCartViewModel.CartViewState> processItemAdded(
+      ShoppingCartDomain.ItemAdded event,
+      Optional<ShoppingCartViewModel.CartViewState> state) {
```

Comment:
we may need to re-think if we want to pass an Optional here. Currently, we support a few variants and use reflection to find out how to call the method. When we generate it, we will need to pick one style. 

To keep it similar to the entities API, we could introduce an empty state for the view. If we don't have anything on the DB, we pass the emptyState to the user. User can decide to use `null` or whatever the profo default instance (or whatever they prefer). 


And we should swap the param: `(State, Command) => Effect`

Reply:
sounds good to me, consistency is an important goal